// jmap/index.js
(function (factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['jquery'], factory);
  } else if (typeof exports === 'object') {
    // Node/CommonJS
    module.exports = factory(require('jquery'));
  } else {
    // Browser globals
    factory(jQuery);
  }
}(function ($) {

  "use strict";

  // Create the defaults once
  var pluginName = "maps",
      defaults = {
        lat: 45.4340257,
        lng: 12.3334331,
        customStyle: [],
        zoom: 12,
        mapType: google.maps.MapTypeId.ROADMAP,
        marker: {
          title: '',
          icon: "/images/__wd__/marker.png"
        }
      };

  // The actual plugin constructor
  function Plugin ( element, options ) {
    this.element   = element;
    this.settings  = $.extend( {}, defaults, options );
    this._defaults = defaults;
    this._name     = pluginName;
    this.init();
  }

  // Avoid Plugin.prototype conflicts
  $.extend(Plugin.prototype, {
    init: function () {

      var myPosition = new google.maps.LatLng(this.settings.lat, this.settings.lng),
      mapOptions = {
          zoom: this.settings.zoom,
          center: myPosition,
          mapTypeId: this.settings.mapType,
          mapTypeControl: false,
          panControl: false,
          rotateControl: false,
          scaleControl: false,
          streetViewControl: false,
          scrollwheel: false,
          styles: this.settings.customStyle
        };
      var map = new google.maps.Map(document.getElementById('map'), mapOptions);
      var marker = new google.maps.Marker({ 
        'position': myPosition,
        'map': map,
        'title': this.settings.marker.title,
        'icon': this.settings.marker.icon
      });

    }
  });

  $.fn[ pluginName ] = function ( options ) {
    return this.each(function() {
      if ( !$.data( this, "mm_" + pluginName ) ) {
        $.data( this, "mm_" + pluginName, new Plugin( this, options ) );
      }
      // Per chiamare i metodi del plugin usare questa sintassi:
      // $('#elemento').data('mm_maps').toggleMenu();
    });
  };

}));
